config/routes.rb

```diff
@@ -11,6 +15,12 @@
     resources :posts, only: [:index, :show]
   end
 
+  resources :sitemaps, :only => :index
+  get "sitemap.xml" => "sitemaps#index", format: "xml", as: :sitemap
+
+  resources :feeds, :only => :index
+  get "feeds.rss" => "feeds#index", format: "rss", as: :feed
```

Comment:
You seem to have routed each of these three different ways.  The resources way should be the winner.